c7n/resources/iam.py

```diff
@@ -302,6 +302,21 @@ def _user_policies(resource):
         return matched
 
 
+@User.filter_registry.register('access-key-last-used')
+class UserAccessKeyLastUsed(Filter):
+    schema = type_schema(
+        'access-key-last-used',
+        keyid={'type': 'string'})
+
+    def process(self, resources, event=None):
+        keyid = self.data.get('keyid', None)
+        if not keyid:
+            return []
+        client = local_session(self.manager.session_factory).client('iam')
+        return client.get_access_key_last_used(AccessKeyId=keyid)
```

Comment:
agreed. thought of that while driving home yesterday. will make adjustments to ensure it works as a filter is supposed to.